Sources/SwiftParser/Lookahead.swift

```diff
@@ -171,6 +171,15 @@ extension Parser.Lookahead {
           return
         }
       }
+      if case .isolated = attr {
+        guard
+          self.consume(if: .leftParen) != nil,
+          self.consume(if: .identifier) != nil,
+          self.consume(if: .rightParen) != nil
+        else {
+          return
+        }
+      }
```

Comment:
You should be able to merge this with `if case .convention = attr` by changing it to `switch attr { case .convention, .isolated`

Reply:
I could, but the `@convention` handling is wrong and will need to be split out when it's fixed.

Reply:
Oh, good catch. Filed https://github.com/apple/swift-syntax/issues/2474 and fixing it in https://github.com/apple/swift-syntax/pull/2475.

You should be able to tag along with that PR and just call `skipSingle` for `@isolated` parsing. That should also cover any future additions to the `@isolated` grammar.

Reply:
Rebased onto your commit.